### ISSUE TITLE
[BUG/Help] 为什么在ptuning的main.py中input默认是要拼接prompt和response，为什么这么设计，还是我理解得不太对？

### ISSUE THREAD

**Issue by NeilRon** [THUDM/ChatGLM2-6B#183]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

```python
def preprocess_function_train(examples):
    # ......
    prompt = prefix + prompt
    a_ids = tokenizer.encode(text=prompt, add_special_tokens=True, truncation=True,
                             max_length=data_args.max_source_length)
    b_ids = tokenizer.encode(text=answer, add_special_tokens=False, truncation=True,
                             max_length=data_args.max_target_length)

    context_length = len(a_ids)
    input_ids = a_ids + b_ids + [tokenizer.eos_token_id]
    labels = [tokenizer.pad_token_id] * context_length + b_ids + [tokenizer.eos_token_id]
    
    pad_len = max_seq_length - len(input_ids)
    input_ids = input_ids + [tokenizer.pad_token_id] * pad_len
    labels = labels + [tokenizer.pad_token_id] * pad_len
    # ......
```
从代码上看input_ids拼接了prompt列和answer列，labels拼接了和prompt等长的padding字符再加上answer列，从打印的日志看也是这样的， 为什么这么设计呢？

### Steps To Reproduce

RT

### Environment

```markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```

**Comment by cyxdsp** [comment]
你要不去群里问问

**Comment by QA-long** [comment]
他本质上就是一个续写模型，拼在一起很合理啊。

**Comment by lilongxian** [comment]
官方设计这是很正确的。这是通用nlp自监督学习常用的数据建模方法，对应损失函数一般设计思路：
loss=crossEntropy(logist[：，-1], label_ids[：, 1:])   
loss = loss*mask/sum(mask) 
logist即input_ids+position信息经模型网络编码后的输出3d张量。
1. 使用与预训练（label用Input_ids表示，mask可以不需要，对整体句子上下文学习，也可以使用句子级mask，如mpnet、glm1-6b那样保留mask句的位置和mask句内部token位置信息，在保证整体句子长句不变的情况下来预测mask句）。
2. 使用于微调训练中，整体句子input: a句 [gmask][sop] response句 [eop] 经模型学习，，网络编码输出logist 3D张量，使用上述loss函数计算损失，保证整体句子长度不变，就需要将label中的非response部分mask掉，在此基础上才可以仅计算response部分位置上模型的tokens编码效果，计算其loss，所以label_ids即要mask掉前面的input部分，还要把其位置填补完整，充当mask。

**Comment by NeilRon** [comment]
> 他本质上就是一个续写模型，拼在一起很合理啊。

不太理解，续写模型会把答案拼在prompt里么？prompt中不是应该拼[MASK]或者[gMASK]之类的标识么？

**Comment by RobbertOnly** [comment]
文本填空和文本生成是nlp的两个任务，建议看看nlp基础知识。
